kernel/input.go

```diff
@@ -127,11 +126,14 @@ func (k InputKernel) Converge(data []byte) (interface{}, thema.TranslationLacuna
 		return nil, nil, err
 	}
 
-	// Validate that the data constitutes an instance of at least one of the schemas in the lineage
-	inst := k.lin.ValidateAny(v)
```

Comment:
So this call needs to be preserved - it just needs to happen after the `Validate()` call that you added.

Reasoning is that we still want to check against all possible schema inputs (which `ValidateAny()` does), we just want to try against the `To` version first, and use the error from it in the event of failure against all the schemas to improve feedback.

Reply:
Not sure I follow the logic if after the one I added, but pushed a change that that puts in before.... maybe that is what you meant?

Reply:
it's backwards to what i intended here, but i don't think it's a big deal either way - whether it's preferable to do what you did or what i did depends on speculation (whether most users use 

A proper job would stop using `ValidateAny()` and instead construct [the loop](https://github.com/grafana/thema/blob/main/impl.go#L218-L222) ourselves, so that we never visit a schema more than once, and can choose to visit the `To` schema first. (There's a public equivalent to everything that's private in the link, so it's doable)

But I'm fine with this as written, or the way i originally described, or with creating our own loop :) Don't feel like overindexing on this rn